packages/devtools_app/lib/src/bloc/bloc/bloc_list_bloc.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../eval_on_dart_library.dart';
+import '../../globals.dart';
+import '../models/bloc_object.dart';
+
+part 'bloc_list_event.dart';
+part 'bloc_list_state.dart';
+
+class BlocListBloc extends Bloc<BlocListEvent, BlocListState> {
+  BlocListBloc(this._evalOnDartLibrary) : super(BlocListLoadInProgress()) {
+    _postEventSubscription =
+        serviceManager.service.onExtensionEvent.where((event) {
+      return event.extensionKind == 'bloc:bloc_map_changed';
+    }).listen((_) {
+      add(BlocListRequested());
+    });
+    add(BlocListRequested());
+  }
+
+  final EvalOnDartLibrary _evalOnDartLibrary;
+  StreamSubscription<void> _postEventSubscription;
+
+  @override
+  Future<void> close() {
+    _postEventSubscription.cancel();
+    return super.close();
+  }
+
+  @override
+  Stream<BlocListState> mapEventToState(
+    BlocListEvent event,
+  ) async* {
+    if (event is BlocListRequested) {
+      yield* _mapBlocListRequestedToState(event);
+    } else if (event is BlocSelected) {
+      yield* _mapBlocSelectedToState(event);
+    }
+  }
+
+  Stream<BlocListState> _mapBlocListRequestedToState(
+      BlocListRequested event) async* {
+    yield BlocListLoadInProgress();
+    try {
+      final List<BlocObject> blocList = await _getBlocList();
```

Comment:
```suggestion
      final blocList = await _getBlocList();
```

We shouldn't need to specify the type explicitly since it is inferred from `_getBlocList`